extensions/amp-accordion/1.0/accordion.js

```diff
@@ -397,16 +415,53 @@ export function AccordionContent({
 }) {
   const ref = useRef(null);
   const hasMountedRef = useRef(false);
-  const {contentId, headerId, expanded, animate, setContentId} = useContext(
-    SectionContext
-  );
+  const {
+    contentId,
+    headerId,
+    expanded,
+    animate,
+    setContentId,
+    expandHandler,
+    experimentDisplayLocking,
+  } = useContext(SectionContext);
   const classes = useStyles();
+  const [contentVisibility, setContentVisibility] = useState(false);
+  const enableDisplayLocking = contentVisibility && experimentDisplayLocking;
 
   useEffect(() => {
     hasMountedRef.current = true;
     return () => (hasMountedRef.current = false);
   }, []);
 
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+
+    const element = ref.current;
+    const win = element.ownerDocument.defaultView;
+    if (!win) {
+      return;
+    }
+
+    const supportsContentVisibility = win.CSS.supports(
+      'content-visibility',
+      'hidden-matchable'
+    );
+    setContentVisibility(supportsContentVisibility);
+    if (!experimentDisplayLocking || !supportsContentVisibility) {
+      return;
+    }
+
+    const beforeMatchHandler = () => {
+      if (!element.parentElement.hasAttribute('expanded')) {
```

Comment:
Seems weird to dig into the parent element here. Can we instead derive the logic from our own state? Or this element?

Reply:
I used `aria-hidden` in the latest revision.  Using classlist required a dependency on the class names generated by JSX, so opted against that, but open to suggestion.